services/queued/internal/queue/queue.go

```diff
@@ -228,249 +234,279 @@ func zipOutputToBase64(output_path string) (string, error) {
 	return base64.StdEncoding.EncodeToString(data), nil
 }
 
-func processMessages(wg *sync.WaitGroup) {
+// Exit causes the virtual machine to exit. Exit never returns to the caller.
+func Exit() {
+	var err error
+	err = syscall.Reboot(syscall.LINUX_REBOOT_CMD_POWER_OFF)
+	if err != nil {
+		log.Printf("power off failed: %v", err)
+	}
+	err = syscall.Reboot(syscall.LINUX_REBOOT_CMD_HALT)
+	if err != nil {
+		log.Printf("halt failed: %v", err)
+	}
+	os.Exit(1)
+	select {}
+}
+
+func ProcessMessages(wg *sync.WaitGroup, ctx context.Context) {
+
 	defer wg.Done() // let main know we are done cleaning up
 
+	var lastGoodRead = time.Now()
+
+	args := GetArguments()
+
 	for {
-		msg, more := <-queue
-		if !more {
-			log.Println("<6>Message queue empty and closed.")
+
+		select {
+
+		case <-ctx.Done(): // if cancel() execute
+			log.Println("Gracefully canceling ProcessMessages loop...")
 			return
-		}
 
-		log.Println("<6>Start processing of message.")
+		default:
 
-		func() {
-			// This is to always log last request, in case something went terribly bad
-			defer func() {
-				r := recover()
-				if r != nil {
-					log.Panicf("<3>Panicing on message %v", msg.MetaData) // just fail quickly
-				} else {
-					log.Println("<6>Finished processing of message.")
-				}
-			}()
+			log.Println("Get message via http request")
 
-			var outputVolume docker.DockerVolume
-			// we do not want to delte the tmp files, it may be useful,
-			// also the line below will not do the trick, since args
-			// are bound when defer is called, not when the function is
-			// called, so it will call Cleanup on empty path :(
-			//defer outputVolume.Cleanup()
-
-			result := func() string {
-				ctx := context.Background()
-
-				config := docker.DockerConfig{
-					Volumes:     []docker.DockerVolume{},
-					DockerImage: msg.DockerImage,
-					Timeout:     uint16(msg.MaxRunTime),
-					Memory:      int64(msg.Memory * 1024 * 1024),
-					Username:    arguments.DockerUsername,
-					Password:    arguments.DockerPassword,
-				}
+			resp, err := http.Get(args.MqReadServiceAddr)
+			if err != nil {
+				log.Println(err)
+				continue
+			}
 
-				docker.PullImage(ctx, config)
+			msg := requestMessage{
+				MaxRunTime: 500, // 300 for test and 200 for build
+				Memory:     2048,
+			}
 
-				env, err := docker.GetDockerEnv(ctx, msg.DockerImage)
-				if err != nil {
-					return err.Error()
-				}
+			var isOk = true
 
-				outputVolume = getVolume(strings.TrimSpace(env["OUTPUT_PATH"]), false)
+			func() {
+				defer func() {
+					resp.Body.Close()
+				}()
 
-				submissionVolume := getVolume(strings.TrimSpace(env["SUBMISSION_PATH"]), true)
-				defer submissionVolume.Cleanup()
+				log.Println("Status code returned from HTTP GET ", resp.StatusCode)
```

Comment:
You should add <N> to logs so levels are parsed correctly.

Reply:
No completely sure what exactly you mean... Can you elaborate?

Reply:
Fixed after consultation on Teams.